### ISSUE TITLE
Improve retry policy

### ISSUE THREAD

**Issue by kubmichael** [gakonst/ethers-rs#2119]
**Is your feature request related to a problem? Please describe.**
I came to this problem by using [anvil](https://github.com/foundry-rs/foundry). During communication between anvil and chain provider, connection reset and abort happens and it is look like that anvil isn't retrying on such errors.  My proposal that retry policy should be triggered also on some tcp errors, I am talking about this part - [code](https://github.com/gakonst/ethers-rs/blob/7e7f9041b3f5a601a8fca8ccbce0287518f8cc33/ethers-providers/src/transports/retry.rs#L444)

**Describe the solution you'd like**
Solution can be similar to reqwest-middleware crate, something like this - [link](https://github.com/TrueLayer/reqwest-middleware/blob/f854725791ccf4a02c401a26cab3d9db753f468c/reqwest-retry/src/retryable.rs#L53)

Thanks

**Comment by mattsse** [comment]
what errors do you have in mind?
retry only retries errors that could be attributed to a spurious network or rate-limit.

**Comment by kubmichael** [comment]
> what errors do you have in mind? retry only retries errors that could be attributed to a spurious network or rate-limit.
Yep, correct, but I think those errors are missed: io::ErrorKind::ConnectionReset and io::ErrorKind::ConnectionAborted, I think they are under  reqwest_err.is_request()
